### PR TITLE
[diag] use CmdLineParser to parse numbers in diag commands

### PR DESCRIPTION
**Pull request by zhanglongxia**

### CODE REVIEW

File: src/core/diags/factory_diags.cpp [hunk 1]
```diff
@@ -354,30 +350,31 @@ Error Diags::ProcessRepeat(uint8_t aArgsLength, char *aArgs[])
...
         }
         else
         {
             ExitNow(error = kErrorInvalidArgs);
         }
 
-        VerifyOrExit(value <= OT_RADIO_FRAME_MAX_SIZE, error = kErrorInvalidArgs);
-        VerifyOrExit(value >= OT_RADIO_FRAME_MIN_SIZE, error = kErrorInvalidArgs);
-        mTxLen = static_cast<uint8_t>(value);
+        VerifyOrExit(txLength <= OT_RADIO_FRAME_MAX_SIZE, error = kErrorInvalidArgs);
```
Comment (Irving-cl) [inline src/core/diags/factory_diags.cpp:375]: Can we write this in one line?

**Review by Irving-cl — APPROVED** [review]
LGTM 👍

**Comment by bukepo** [comment]
why is the RCP size increased?

**Review by abtink — APPROVED** [review]
LGTM. 👍

**Comment by zhanglongxia** [comment]
> why is the RCP size increased?

The original code uses the function `strtol ` to convert the string to the number. The function `Utils::CmdLineParser::ParseAsXxx` is [implemented](https://github.com/openthread/openthread/blob/main/src/core/utils/parse_cmdline.cpp#L109) by OpenThread itself. The implementation of the `Utils::CmdLineParser::ParseAsXxx` increases the code size.

**Review by bukepo — APPROVED** [review]

**Review by jwhui — APPROVED** [review]

**Pull request merged**